src/App.tsx

```diff
@@ -1,14 +1,67 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
+import { getTodos } from './api';
+import { Todo } from './types/Todo';
+import { FilteringOption } from './types/FilteringOption';
+import { Status } from './types/Status';
 import { TodoList } from './components/TodoList';
-import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
+import { TodoFilter } from './components/TodoFilter';
 import { Loader } from './components/Loader';
 
+const getPreparedTodos = (
+  todos: Todo[],
+  { selectedStatus, query }: FilteringOption,
+) => {
+  let preparedTodos = [...todos];
+
+  switch (selectedStatus) {
+    case Status.active:
+      preparedTodos = todos.filter(todo => !todo.completed);
+      break;
+    case Status.completed:
+      preparedTodos = todos.filter(todo => todo.completed);
+      break;
+  }
+
+  if (query) {
+    const normalizedQuery = query.trim().toLowerCase();
+
+    preparedTodos = preparedTodos.filter(todo => {
+      const normalizedTitle = todo.title.toLowerCase();
+
+      return normalizedTitle.includes(normalizedQuery);
+    });
+  }
+
+  return preparedTodos;
+};
+
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [selectedTodo, setSelectedTodo] = useState<Todo | null>(null);
+
+  const [isLoading, setIsLoading] = useState(false);
+
+  const [query, setQuery] = useState('');
+  const [selectedStatus, setSelectedStatus] = useState<Status>(Status.all);
+
+  const visibleTodos = getPreparedTodos(todos, { selectedStatus, query });
+
+  useEffect(() => {
+    getTodos()
+      .then(setTodos)
+      .finally(() => setIsLoading(true));
```

Comment:
```suggestion
  useEffect(() => {
    setIsLoading(true);
    
    getTodos()
      .then(setTodos)
      .finally(() => setIsLoading(false));
```
When the loading is finished you should hide the loader